src/amber/support/client_reload.cr

```diff
@@ -0,0 +1,94 @@
+require "watcher"
+
+module Amber::Support
+  # Used by `Amber::Pipe::Reload`
+  #
+  # Allow clients browser reloading using WebSockets and file watchers.
+  struct ClientReload
+    WEBSOCKET_PATH = rand(0x10000000).to_s(36)
+
+    SESSIONS = [] of HTTP::WebSocket
+
+    def initialize
+      create_reload_server
+      spawn watch_files
+    end
+
+    private def create_reload_server
+      Amber::WebSockets::Server::Handler.new "/#{WEBSOCKET_PATH}" do |session|
+        SESSIONS << session
+        session.on_close do
+          SESSIONS.delete session
+        end
+      end
+    end
+
+    private def reload_clients(msg)
+      SESSIONS.each do |session|
+        session.@ws.send msg
+      end
+    end
+
+    private def watch_files
+      puts "🔃  Your ClientBot is vigilant. beep-boop..."
+      watch(["public/**/*"]) do |event|
+        event.on_change do |files|
+          files.each do |file, timestamp|
+            puts "🔃  watching file: ./#{file}"
+            case file
+            when .ends_with? ".css"
+              reload_clients(msg: "refreshcss")
+            else
+              reload_clients(msg: "reload")
+            end
+          end
+        end
+      end
+    end
+
+    # Code from https://github.com/tapio/live-server/blob/master/injected.html
+    INJECTED_CODE = <<-HTML
+    <!-- Code injected by Amber Framework -->
+    <script type="text/javascript">
+      // <![CDATA[  <-- For SVG support
+      if ('WebSocket' in window) {
+        (function() {
+          function refreshCSS() {
+            console.log('Reloading CSS...');
+            var sheets = [].slice.call(document.getElementsByTagName('link'));
+            var head = document.getElementsByTagName('head')[0];
+            for (var i = 0; i < sheets.length; ++i) {
+              var elem = sheets[i];
+              head.removeChild(elem);
```

Comment:
Should we only remove and add the link if its a `stylesheet` or should we do this for all links?

Reply:
I just copied a working code from nodejs package `live-server`, What you suggest we could use? 😅

Reply:
The logic here gets all the `links`, removes it from the head, determines if its a `stylesheet` on line 62.  If it is, it adds a `_cacheOverride={Date()}` to the link to break the cache.  Then it adds the link back into the head.

What I'm wondering if we need to remove it then append it back on #67 if its not a stylesheet.  Why remove it and add it back if your not going to add a `cacheOverride`?

What if we only remove and add it back if its a stylesheet?  Something like this:
```
if (elem.href && typeof rel != 'string' || rel.length == 0 || rel.toLowerCase() == 'stylesheet') {
  head.removeChild(elem);
  var url = elem.href.replace(/(&|\\?)_cacheOverride=\\d+/, '');
  elem.href = url + (url.indexOf('?') >= 0 ? '&' : '?') + '_cacheOverride=' + (new Date().valueOf());
  head.appendChild(elem);
}
```

Reply:
Done!